test/Rings-conformance-tests.jl

```diff
@@ -147,50 +147,104 @@ function test_mutating_op_like_neg(f::Function, f!::Function, A)
 end
 
 function test_mutating_op_like_add(f::Function, f!::Function, A, B)
-   # initialize storage var with different values to check that its value is not used
-   for z in [zero(A), deepcopy(A), deepcopy(B)]
-      a = deepcopy(A)
-      b = deepcopy(B)
-      z = f!(z, a, b)
-      @test equality(z, f(A, B))
-      @test a == A
-      @test b == B
-   end
-
-   a = deepcopy(A)
-   b = deepcopy(B)
-   a = f!(a, a, b)
-   @test equality(a, f(A, B))
-   @test b == B
-
-   a = deepcopy(A)
-   b = deepcopy(B)
-   b = f!(b, a, b)
-   @test equality(b, f(A, B))
-   @test a == A
-
-   a = deepcopy(A)
-   b = deepcopy(B)
-   a = f!(a, b, b)
-   @test equality(a, f(B, B))
-   @test b == B
-
-   b = deepcopy(B)
-   b = f!(b, b, b)
-   @test equality(b, f(B, B))
+  # initialize storage var with different values to check that its value is not used
+  for z in [zero(A), deepcopy(A), deepcopy(B)]
+     a = deepcopy(A)
+     b = deepcopy(B)
+     z = f!(z, a, b)
+     @test equality(z, f(A, B))
+     @test a == A
+     @test b == B
+  end
+
+  a = deepcopy(A)
+  b = deepcopy(B)
+  a = f!(a, a, b)
+  @test equality(a, f(A, B))
+  @test b == B
+
+  a = deepcopy(A)
+  b = deepcopy(B)
+  b = f!(b, a, b)
+  @test equality(b, f(A, B))
+  @test a == A
+
+  a = deepcopy(A)
+  b = deepcopy(B)
+  a = f!(a, b, b)
+  @test equality(a, f(B, B))
+  @test b == B
+
+  b = deepcopy(B)
+  b = f!(b, b, b)
+  @test equality(b, f(B, B))
+
+  a = deepcopy(A)
+  b = deepcopy(B)
+  a = f!(a, b)
+  @test equality(a, f(A, B))
+  @test b == B
+
+  b = deepcopy(B)
+  b = f!(b, b)
+  @test equality(b, f(B, B))
+end
 
-   a = deepcopy(A)
-   b = deepcopy(B)
-   a = f!(a, b)
-   @test equality(a, f(A, B))
-   @test b == B
-
-   b = deepcopy(B)
-   b = f!(b, b)
-   @test equality(b, f(B, B))
+function test_mutating_op_like_mul(f::Function, f!::Function, A, B; left_factor_is_scalar::Bool = false, right_factor_is_scalar::Bool = false)
```

Comment:
Alternative design picking up on something we discussed during initial design of `test_mutating_ops_*` when I wanted to restricted the type of the outputs: we could indicate the "target type" being tested by an optional argument. Like so perhaps:
```suggestion
function test_mutating_op_like_mul(f::Function, f!::Function, A, B, T = Any)
```
Then we can restrict to those tests where the target `isa T`.

To me the result reads much more naturally.


But I can also live with the current design... let's think a bit about it maybe, dunno.